tlslite/tlsconnection.py

```diff
@@ -3649,12 +3649,14 @@ def _sigHashesToList(settings, privateKey=None, certList=None,
                         continue
                     # advertise support for both rsaEncryption and RSA-PSS OID
                     # key type
-                    sigAlgs.append(getattr(SignatureScheme,
-                                           "rsa_{0}_rsae_{1}"
-                                           .format(schemeName, hashName)))
-                    sigAlgs.append(getattr(SignatureScheme,
-                                           "rsa_{0}_pss_{1}"
-                                           .format(schemeName, hashName)))
+                    if certType != 'rsa-pss':
```

Comment:
Looks good, though the loop does look it assumes that RSA is the only thing that is supported.

Reply:
yes, it does do that, but it's expected as ECDSA is not supported